packages/components/src/edit-in-place-control/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef, useEffect } from '@wordpress/element';
+import { ENTER, ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+
+function EditInPlaceControl( { label = '', onClick = noop, onChange = noop } ) {
+	const [ edit, setEdit ] = useState( false );
+	const [ value, setValue ] = useState( label );
+	const prevValue = useRef();
+
+	const inputRef = useRef();
+	const buttonRef = useRef();
+
+	useEffect( () => {
+		prevValue.current = value;
+		if ( edit ) {
+			inputRef.current.focus();
+			inputRef.current.select();
+		} else {
+			buttonRef.current.focus();
+		}
+	}, [ edit ] );
+
+	return (
+		<>
+			{ ! edit && (
+				<Button
+					ref={ buttonRef }
+					className="components-edit-in-place-control__label"
+					onClick={ () => {
+						setEdit( true );
+						onClick();
```

Comment:
We should probably also pass down the `event` argument.

Reply:
It's just a helpful little addition so that the users can still pass down `onClick` function if they need custom click handlers attached to the component. It's probably not needed or doesn't make much sense in this case though, since this component is not just a wrapper of a button but also could be a wrapper of an input, which would be pretty confusing if we try to attach an `onClick` handler there.